nuxeo-retention/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -279,68 +305,100 @@ public void evalRules(Map<String, Set<String>> docsToCheckAndEvents) {
         Framework.getService(WorkManager.class).schedule(work, WorkManager.Scheduling.ENQUEUE);
     }
 
-    protected ELActionContext initActionContext(DocumentModel doc, CoreSession session) {
-        ELActionContext ctx = new ELActionContext(new ExpressionContext(), new ExpressionFactoryImpl());
-        ctx.setCurrentPrincipal(session.getPrincipal());
-        doc.detach(true);
-        ctx.setCurrentDocument(doc);
-        return ctx;
-    }
-
-    protected boolean evaluateConditionExpression(ELActionContext ctx, String expression) {
-        Calendar now = Calendar.getInstance();
+    protected boolean evaluateConditionExpression(Record record, String eventInput, CoreSession session) {
+        String expression = record.getRule(session).getStartingPointExpression();
         if (StringUtils.isEmpty(expression)) {
-            return false;
+            return true;
         }
+        ELActionContext ctx = new ELActionContext(new ExpressionContext(), new ExpressionFactoryImpl());
+        record.getDocument().detach(true);
+        ctx.setCurrentDocument(record.getDocument());
+        Calendar now = Calendar.getInstance();
         ctx.putLocalVariable("currentDate", now);
+        if (StringUtils.isNotBlank(eventInput)) {
+            if (EVENT_INPUT_PATTERN.matcher(eventInput).matches()) {
+                ctx.putLocalVariable(EVENT_BASED_RULE_INPUT, eventInput);
+            } else {
+                log.warn(
+                        "Invalid retention event input: {}, only alphanumeric, dash, underscore and period character are allowed",
```

Comment:
"... is not following the expected pattern" will be easier to maintain if the regex is updated.